test/installers/mongodb_installer.go

```diff
@@ -0,0 +1,160 @@
+// Copyright 2021 Red Hat, Inc. and/or its affiliates
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package installers
+
+import (
+	"fmt"
+
+	"github.com/kiegroup/kogito-cloud-operator/core/client/kubernetes"
+	"github.com/kiegroup/kogito-cloud-operator/test/framework"
+	mongodbv1 "github.com/mongodb/mongodb-kubernetes-operator/pkg/apis/mongodb/v1"
+	coreapps "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	rbac "k8s.io/api/rbac/v1"
+	apiextensionsv1beta1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1beta1"
+)
+
+var (
+	// mongoDbYamlNamespacedInstaller installs MongoDB namespaced using YAMLs
+	mongoDbYamlNamespacedInstaller = YamlNamespacedServiceInstaller{
+		installNamespacedYaml:           installMongoDbUsingYaml,
+		waitForNamespacedServiceRunning: waitForMongoDbUsingYamlRunning,
+		getAllNamespaceYamlCrs:          getMongoDbCrsInNamespace,
+		uninstallNamespaceYaml:          uninstallMongoDbUsingYaml,
+		namespacedYamlServiceName:       mongoDBOperatorServiceName,
+	}
+
+	mongoDBOperatorServiceName    = "Mongo DB"
+	mongoDBOperatorVersion        = "v0.2.2"
+	mongoDBOperatorDeployFilesURI = "https://raw.githubusercontent.com/mongodb/mongodb-kubernetes-operator/" + mongoDBOperatorVersion + "/deploy/"
+)
+
+// GetMongoDbInstaller returns MongoDB installer
+func GetMongoDbInstaller() ServiceInstaller {
+	return &mongoDbYamlNamespacedInstaller
+}
+
+func installMongoDbUsingYaml(namespace string) error {
+	framework.GetLogger(namespace).Info("Deploy MongoDB from yaml files", "file uri", mongoDBOperatorDeployFilesURI)
+
+	if !framework.IsMongoDBAvailable(namespace) {
+		if err := framework.LoadResource(namespace, mongoDBOperatorDeployFilesURI+"crds/mongodb.com_mongodb_crd.yaml", &apiextensionsv1beta1.CustomResourceDefinition{}, nil); err != nil {
+			return err
+		}
+	}
+
+	if err := framework.LoadResource(namespace, mongoDBOperatorDeployFilesURI+"service_account.yaml", &corev1.ServiceAccount{}, nil); err != nil {
```

Comment:
`framework.LoadResource` also set the namespace for created resource. In this case it wouldn't be a problem as namespace is not set in YAMLs. Also `framework.LoadResource` allows adjustment of resources before they are applied (that will come handy for latest MongoDB operator once we will update, the installation approach is significantly refactored).

I think it could be ok to leave the current implementation as it is going to be refactored anyway in near future.